product_variant_default_code/views/product_view.xml

```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" ?>
+<odoo>
+    <record id="product_template_only_form_view" model="ir.ui.view">
+        <field name="model">product.template</field>
+        <field name="inherit_id" ref="product.product_template_only_form_view" />
+        <field name="arch" type="xml">
+            <field name="default_code" position="after">
+                <field
+                    name="code_prefix"
+                    attrs="{'invisible':
+                          [('attribute_line_ids', '=', [])]}"
+                />
+                <field
+                    name="reference_mask"
+                    placeholder="[attribute3]-[attribute1]"
+                    attrs="{'invisible':
+                          [('attribute_line_ids', '=', [])]}"
+                    groups="product_variant_default_code.group_product_default_code_manual_mask"
+                />
+            </field>
+            <xpath expr="//page/field[@name='attribute_line_ids']" position="before">
+                    <field
+                    name="variant_default_code_error"
+                    class="alert alert-danger alert-dismissable"
+                    role="alert"
+                    attrs="{'invisible':[('variant_default_code_error','=', False)]}"
```

Comment:
This is useless to show if there aren't variants/attributes:

![image](https://user-images.githubusercontent.com/5040182/236150917-26c1cd06-133e-44c3-969e-605de23dc33b.png)

Reply:
solved